contracts/system_tests/TestNFTRewardsFlow.sol

```diff
@@ -0,0 +1,166 @@
+      weight: 1000 * 10**18,
+      discountRate: 450000000,
+      ballot: IJBFundingCycleBallot(address(0))
+    });
+
+    _fundingCycleMetadata = JBFundingCycleMetadata({
+      global: JBGlobalFundingCycleMetadata({allowSetTerminals: false, allowSetController: false}),
+      reservedRate: _reservedRate,
+      redemptionRate: 5000, //50%
+      ballotRedemptionRate: 0,
+      pausePay: false,
+      pauseDistributions: false,
+      pauseRedeem: false,
+      pauseBurn: false,
+      allowMinting: true,
+      allowChangeToken: true,
+      allowTerminalMigration: false,
+      allowControllerMigration: false,
+      holdFees: false,
+      useTotalOverflowForRedemptions: false,
+      useDataSourceForPay: true,
+      useDataSourceForRedeem: false,
+      dataSource: address(0)
+    });
+
+    _projectOwner = multisig();
+
+    _terminals.push(jbETHPaymentTerminal());
+
+    _projectId = _controller.launchProjectFor(
+      _projectOwner,
+      _projectMetadata,
+      _fundingCycleData,
+      _fundingCycleMetadata,
+      block.timestamp,
+      _groupedSplits,
+      _fundAccessConstraints,
+      _terminals,
+      ''
+    );
+
+    evm.prank(_projectOwner);
+    _nft = _controller.issueToken721For(
+      _projectId,
+      'NFT',
+      'N',
+      'ipfs://',
+      IJBToken721UriResolver(address(0)),
+      'ipfs://'
+    );
+
+    _jbNFTRewardDataSourceDelegate = new JBNFTRewardDataSourceDelegate(
+      _projectId,
+      _controller,
+      _nft,
+      100,
+      JBTokenAmount({
+        token: address(0),
+        value: 1 * 10**18,
+        decimals: 18,
+        currency: JBCurrencies.ETH
+      })
+    );
+
+    _fundingCycleData = JBFundingCycleData({
+      duration: 60 * 60,
+      weight: 1000 * 10**18,
+      discountRate: 450000001,
+      ballot: IJBFundingCycleBallot(address(0))
+    });
+
+    _fundingCycleMetadata = JBFundingCycleMetadata({
+      global: JBGlobalFundingCycleMetadata({allowSetTerminals: false, allowSetController: false}),
+      reservedRate: _reservedRate,
+      redemptionRate: 5000, //50%
+      ballotRedemptionRate: 0,
+      pausePay: false,
+      pauseDistributions: false,
+      pauseRedeem: false,
+      pauseBurn: false,
+      allowMinting: true,
+      allowChangeToken: true,
+      allowTerminalMigration: false,
+      allowControllerMigration: false,
+      holdFees: false,
+      useTotalOverflowForRedemptions: false,
+      useDataSourceForPay: true,
+      useDataSourceForRedeem: false,
+      dataSource: address(_jbNFTRewardDataSourceDelegate)
+    });
+
+    vm.warp(block.timestamp + 10000);
+
+    evm.prank(_projectOwner);
+    _controller.reconfigureFundingCyclesOf(
+      _projectId,
+      _fundingCycleData,
+      _fundingCycleMetadata,
+      block.timestamp,
+      _groupedSplits,
+      _fundAccessConstraints,
+      ''
+    );
+  }
+
+  function testNFTRewardMint() public {
+    _terminals[0].pay{value: 2 * 10**18}(
+      _projectId,
+      1 * 10**18,
```

Comment:
amount can be whatever when a msg.value is passed, it is however a bit disturbing when they differ;)
+ I think this could be fuzzed